test/ci/MarkdownPreviewTest.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Test the Markdown-preview plugin
+ */
+
+import * as assert from "assert"
+import * as Oni from "oni-api"
+
+interface IPlugin {
+    /**
+     * Gets the current value of a property.
+     * The return type is unknown.
+     *
+     * param: name  A name of a property
+     * returns      The current value of the specified property
+     */
+    getStatus(name: string): any
+}
+
+interface IPluginManager {
+    getPlugin(name: string): Promise<IPlugin | null>
+}
+
+interface IOniWithPluginApi {
+    plugins: IPluginManager
+}
+
+export async function test(typedOni: Oni.Plugin.Api) {
+    const typelessOni = typedOni as any
+    const oni = typelessOni as IOniWithPluginApi
+
+    const markdownPlugin = await oni.plugins.getPlugin("oni-plugin-markdown-preview")
```

Comment:
Is this the API you were asking about? If so, I'm totally on board. 

This is something I've wanted for a while - the long-term vision for the plugins is to have a way for them to export functions that can be used by other plugins, or in a user's configuration, etc. Having a `plugins.getPlugin("<plugin-id>")` seems like a reasonable way to accomplish this. I'd even like to move some of Oni's core functionality to plugins like this - for example, it'd be cool to move our configuration story over to a module/plugin, and have `Oni.configuration` just be a convenience method for `Oni.plugins.getPlugin("oni-configuration")`. Not something I'd expect to be implemented as part of this - just wanted to share some related thoughts 👍